sklearn/datasets/_base.py

```diff
@@ -61,17 +62,28 @@ def clear_data_home(data_home=None):
 
     Parameters
     ----------
-    data_home : str | None
-        The path to scikit-learn data dir.
+    data_home : str, default=None
+        The path to scikit-learn data directory.
     """
     data_home = get_data_home(data_home)
     shutil.rmtree(data_home)
 
 
 def _convert_data_dataframe(caller_name, data, target,
-                            feature_names, target_names):
+                            feature_names, target_names, sparse_data=False):
     pd = check_pandas_support('{} with as_frame=True'.format(caller_name))
-    data_df = pd.DataFrame(data, columns=feature_names)
+    if not sparse_data:
+        data_df = pd.DataFrame(data, columns=feature_names)
+    else:
+        if LooseVersion(pd.__version__) < '0.25':
+            raise ValueError(
+                "Loading sparse datasets as a DataFrame requires Pandas "
+                "v0.25+."
+            )
```

Comment:
I'm more than happy with this solution, but I remember @GaelVaroquaux had some concerns regarding _soft dependencies_ which are hard to know and one only learns about them only if they encounter the issue. I'd be also happy with bumping our min pandas support to 0.25 (which I think is pretty new).

Reply:
I don't have a strong opinion here.

Reply:
IIRC, the minimum pandas version is `0.25` (https://github.com/scikit-learn/scikit-learn/blob/bb997dc5e3fd41e4ac3e7af01511d64abb162fed/sklearn/_build_utils/min_dependencies.py#L31).

Therefore, we could avoid this `if` statement here?